GitCommands/AppTitleGenerator.cs

```diff
@@ -36,19 +34,22 @@ public AppTitleGenerator(IRepositoryDescriptionProvider repositoryDescriptionPro
         /// <param name="workingDir">Path to repository.</param>
         /// <param name="isValidWorkingDir">Indicates whether the given path contains a valid repository.</param>
         /// <param name="branchName">Current branch name.</param>
-        public string Generate(string workingDir, bool isValidWorkingDir, string branchName)
+        public string Generate(string workingDir = null, bool isValidWorkingDir = false, string branchName = null)
         {
             if (string.IsNullOrWhiteSpace(workingDir) || !isValidWorkingDir)
             {
-                return DefaultTitle;
+                return "Git Extensions";
             }
 
-            string repositoryDescription = _repositoryDescriptionProvider.Get(workingDir);
-            var title = string.Format(RepositoryTitleFormat, repositoryDescription, (branchName ?? "no branch").Trim('(', ')'));
+            branchName = branchName?.Trim('(', ')') ?? "no branch";
+
+            var description = _description.Get(workingDir);
+
 #if DEBUG
-            title += " -> DEBUG <-";
+            return $"{description} ({branchName}) - Git Extensions [DEBUG]";
```

Comment:
Great change!
Do we want to see the branch name in release builds though?
(RussKie wanted the version some months ago, I rather keep this clean in releases).

Reply:
Would be nice to have the git commit in the about box.

Reply:
You mean like I suggested in #2533?

Reply:
This looks like an easy way to add commit ID to the assembly metadata:

https://github.com/MarkPflug/MSBuildGitHash